src/identity/identity-wallet.ts

```diff
@@ -1122,4 +1288,136 @@ export class IdentityWallet implements IIdentityWallet {
       isStateGenesis: false
     });
   }
+
+  /** {@inheritdoc IIdentityWallet.transitState} */
+  async transitState(
+    did: DID,
+    oldTreeState: TreeState,
+    isOldStateGenesis: boolean,
+    ethSigner: Signer
+  ): Promise<string> {
+    const newTreeModel = await this.getDIDTreeModel(did);
+    const claimsRoot = await newTreeModel.claimsTree.root();
+    const rootOfRoots = await newTreeModel.rootsTree.root();
+    const revocationRoot = await newTreeModel.revocationTree.root();
+
+    const newTreeState: TreeState = {
+      revocationRoot,
+      claimsRoot,
+      state: newTreeModel.state,
+      rootOfRoots
+    };
+
+    const userId = DID.idFromDID(did);
+
+    let proof;
+    const isEthIdentity = isEthereumIdentity(did); // don't generate proof for ethereum identities
+
+    if (!isEthIdentity && this._prover) {
```

Comment:
shouldn't we throw if `!isEthIdentity && !this._prover` ?
maybe check if `!isEthIdentity` and inside if `!this_prover` throw exception ?